app/Features/QueryAnimeSeasonHandlerBase.php

```diff
@@ -49,7 +51,7 @@ public function handle($request): JsonResponse
      * @param ?AnimeTypeEnum $type
      * @return Builder
      */
-    protected abstract function getSeasonItems($request, ?AnimeTypeEnum $type): Builder;
+    protected abstract function getSeasonItems($request, ?AnimeTypeEnum $type, ?AnimeSeasonEnum $season, ?int $year): Builder;
```

Comment:
This is a very odd way of doing it. The base class was meant for to be used in case of "current" and "explicit" season specification context. That's why the first argument is `request`. That signature should not change in this case. This is a code smell.